sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -554,6 +554,28 @@ object FileSourceMetadataAttribute {
     metadata.getBoolean(FILE_SOURCE_METADATA_COL_ATTR_KEY)
   }
 
+  /**
+   * True if the given data type is supported in file source metadata attributes.
+   *
+   * Strings are spark [[UTFString]], but implementations can also provide scala [[String]] and it
+   * will be converted automatically as needed.
+   *
+   * Currently, only a handful of atomic (primitive) types are supported.
+   */
+  def isSupportedType(dataType: DataType): Boolean = dataType match {
+    case _: LongType | _: IntegerType | _: ShortType | _: ByteType => true
+    case _: DoubleType | _: FloatType => true
+    case _: StringType => true
+    case _: TimestampType => true // really just Long
```

Comment:
Should we also add `DateType` (int), `DayTimeIntervalType` (long) and `YearMonthIntervalType` (int)`?

Reply:
Good catch. Fixed by using `Literal`, `PhysicalType`, and `ColumnVectorUtils.populate`.